### DISCUSSION TITLE
Popover above an iFrame in React

### DISCUSSION THREAD

**Discussion by CarlOfHoly** [floating-ui/floating-ui#2596] [Q&A] [answered]
Hi,

I have implemented a basic Popover as explained in the docs, but I have encountered that I cannot close the Popover when clicking outside of the box when clicking on the underlying iFrame (The black section as shown in the picture). I am using the useDismiss function and have set the modal to true for `FloatingFocusManager`.

![image](https://github.com/floating-ui/floating-ui/assets/15364176/67c8f9da-2559-4efc-954e-c498e512273e)

Is there a nice way of solving this issue or do I have to put an event listener on the iFrame to check for clicks?

**Comment by CarlOfHoly** [comment]
Here is a code sandbox which shows what I mean: [Sandbox](https://codesandbox.io/s/trusting-dewdney-swdryn?file=/src/App.tsx)
When opening the Popover, it does not close when clicking on the iFrame

**Comment by atomiks** [comment, marked as answer]
Use `FloatingOverlay` so clicks are detected on that layer instead of arbitrary layers behind it. Outside clicks can't be detected inside sub-documents since the event doesn't bubble up to the one `useDismiss` attached the listener to.

**Reply by CarlOfHoly** [reply]
Thank you for your reply! What should the FloatingOverlay wrap? The iFrame? The entire page underneath the Popover?
I've looked in the docs, but it's not clear to me how to apply it

**Reply by atomiks** [reply]
It can be a sibling of the popover (with lower z-index) or the parent of it.
